iothub_client/tests/iothubtr_amqp_msgr_ut/iothubtr_amqp_msgr_ut.c

```diff
@@ -959,6 +959,79 @@ static AMQP_MESSENGER_HANDLE create_and_start_messenger2(AMQP_MESSENGER_CONFIG*
     return handle;
 }
 
+char* umock_stringify_AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO(const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* value)
+{
+    char* result = (char*)TEST_malloc(1);
+    (void)value;
+    result[0] = '\0';
+    return result;
+}
+
+int umock_are_equal_AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO(const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* left, const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* right)
+{
+    int result;
+
+    if (((left == NULL) && (right != NULL)) || ((right == NULL) && (left != NULL)))
+    {
+        result = 0;
+    }
+    else
+    {
+        if ((left->source != right->source) || (left->message_id != right->message_id))
+        {
+            result = 0;
+        }
+        else
+        {
+            result = 1;
+        }
+    }
+
+    return result;
+}
+
+int umock_copy_AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO(AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* destination, const AMQP_MESSENGER_MESSAGE_DISPOSITION_INFO* source)
+{
+    int result;
+
+    if ((source == NULL) || (destination == NULL))
+    {
+        result = -1;
+    }
+    else
+    {
+        destination->source = (char*)TEST_malloc(sizeof(source->source));
+        if (destination->source == NULL)
+        {
+            // failed to malloc
+            result = -1;
+            TEST_free(destination->source);
+        }
+        else
+        {
+            if (source->source == NULL)
```

Comment:
This whole block makes no sense, and it will fail valgrind with memory leaks because you firt allocated memory above and stored in destination->source, and down here just override destination->source with the pointer value of source->source.

Reply:
refactored via our chat